mne/io/fiff/tests/test_raw_fiff.py

```diff
@@ -429,6 +430,38 @@ def test_concatenate_raws_bads_order():
         concatenate_raws([raw0, raw4])
 
 
+def test_concatenate_raws_order():
+    """Test concatenation of raw instances when the order of chs varies."""
+    raw0 = _create_toy_data(n_channels=2)
+    raw0._data[0] = np.zeros_like(raw0._data[0])  # set one channel zero
+
+    # Create copy and concatenate raws
+    raw1 = raw0.copy()
+    raw_concat = concatenate_raws([raw0.copy(), raw1])
+    assert raw0.ch_names == raw1.ch_names == raw_concat.ch_names == ["0", "1"]
+    ch0 = raw_concat.copy().pick(picks=["0"]).get_data()
+    assert np.all(ch0 == 0)
+
+    # Change the order of the channels and concatenate again
+    raw1.reorder_channels(["1", "0"])
+    assert raw1.ch_names == ["1", "0"]
+    raws = [raw0.copy(), raw1]
+    with pytest.raises(ValueError):
+        # Fails now due to wrong order of channels
+        raw_concat = concatenate_raws(raws)
+
+    with pytest.raises(ValueError):
+        # still fails, because raws is copied and not changed in place
+        match_channel_orders(raws, copy=True)
+        raw_concat = concatenate_raws(raws)
+
+    # Now passes because all raws have the same order
+    match_channel_orders(raws, copy=False)
+    raw_concat = concatenate_raws(raws)
+    ch0 = raw_concat.copy().pick(picks=["0"]).get_data()
+    assert np.all(ch0 == 0)
+
```

Comment:
Here, I decided to test `concatenate_raws` and `match_channel_orders` in a single test.

Reply:
fine to test both in one test I think. Please add `match` clauses to the `with pytest.raises()` calls as mentioned in another comment.

Reply:
Adressed in e76a1bd1a0734afc8ebf7b7e8d317f663f5bf0a4